addons/stock/models/stock_lot.py

```diff
@@ -128,14 +128,15 @@ def _compute_delivery_ids(self):
     @api.depends('quant_ids')
     def _compute_single_location(self):
         for lot in self:
-            location_id = lot.quant_ids.filtered(lambda q: q.quantity > 0).mapped('location_id')
-            lot.location_id = location_id if len(location_id) == 1 else False
+            quant_ids = lot.quant_ids.filtered(lambda q: q.quantity > 0)
```

Comment:
usually we use _ids when the content is a list of ids or quants with s when it's a record set (except for field declaration)